apps/state-demo/src/app/examples/demo-basics/1/Readme.md

```diff
@@ -1,62 +1,264 @@
+
+To do so, we have to extend our class and use the already existing `ComponentState` interface:
+
+```typescript
+import { RxState } from '@rx-angular/state';
+
+// Displayed shape of the list item (this is a reduced version of the server object)
+export interface DemoBasicsItem {
+  id: string;
+  name: string;
+}
+
+interface ComponentState {
+  refreshInterval: number;
+  list: DemoBasicsItem[];
+  listExpanded: boolean;
+}
+
+export class DemoBasicsComponent2 extends RxState<ComponentState> {
 ```
 
-1. Implement RxState Service ComponentState by extending the RxState class.
+Also, a `super` in the constructor is needed as we extend from another class.
 
 ```typescript
-export class DemoBasicsComponent2 extends RxState<ComponentState>
+constructor(...) {
+  super();
+}
 ```
 
-1.1) Select component State
+1.1) Select component State, display and initialize it.
+
+Lets setup a component property `model$` and display it in the template at the very top to debug the state.
+
+`model$` get the full state object assigned from the components state.
+This is done by using the `$` property of the state service.
 
 ```typescript
-model$ = this.select();
+
+@Component({
+  selector: 'demo-basics-1',
+  template: `
+    model$: <pre>{{model$ | async | json}}</pre>
+    <h3>Demo Basic 1 - Setup and Retrieving State</h3>
+    ...
+  `,
+  ...
+})
+export class DemoBasicsComponent2 extends RxState<ComponentState> {
+  model$ = this.$;
+}
 ```
 
 2.1) Initialize component state
 
+As `RxState` is lazy we can decide on initial values.
+We can initialize the state imperatively over `set` or over a observable ant the `cnnect` method.
+
+Here we use `set`. As initial value we pass the already existing `initComponentState` object.
+
 ```typescript
  constructor(...) {
   ...
-  this.setState(initComponentState);
+  this.set(initComponentState);
 }
 ```
 
+We should see the initial state in the template.
+
 2.2) Connect input bindings
 
+As parts of our state gets passed over input bindings we eed to feed this changes to the components state.
+There is one problem, the `set` function decorated with `@Input` is not compos able with RxJS.
+
+We could write Reactive decorators like `@Input$` and alos do this for all other things not reactive in a component.
+
+Or we could simple use the services imperative `set` method to hook into other imperative code in the class.
+
+In this case we hook into the imperative callback of the `refreshInterval` component input binding.
+
+We run a partial update to our state by providing a object containing the new state slice `{refreshInterval: number}`.
+This can be done by using a reduce function `(oldState) => ({refreshInterval: oldState.refreshInterval + 2})` or just the slice itself `{refreshInterval: 2}`.
+As no previous state is needed to calculate the new value we provide the slice itself for the partial update.
+
+Notice `{refreshInterval}` is the short form of `{refreshInterval: refreshInterval}`.
```

Comment:
Do we need this explanation here?

Reply:
Yes as people may not know this short hand